test/Constraints/construction.swift

```diff
@@ -59,8 +59,7 @@ acceptString("\(hello), \(world) #\(i)!")
 Optional<Int>(1) // expected-warning{{unused}}
 Optional(1) // expected-warning{{unused}}
 _ = .none as Optional<Int>
-Optional(.none) // expected-error{{generic parameter 'T' could not be inferred}} expected-note {{explicitly specify the generic arguments to fix this issue}} {{9-9=<Any>}}
-// expected-error@-1 {{cannot infer contextual base in reference to member 'none'}}
+Optional(.none) // expected-error {{cannot infer contextual base in reference to member 'none'}}
```

Comment:
Still need to figure out where this diagnostic went!

Reply:
So it looks like these changes resulted in the type check exploring the "specify base in reference to member" fix first, and it turns out that that's enough to solve the system. Previously, the "specify generic param" fix was attempted first, which wasn't enough on its own to fix the system, and requiring the additional "specify base" fix. Is this considered a regression? It seems strictly better to offer only the fix that will actually fix the system, but I don't know what the policy is here.